demo/src/app/demo-versions.component.ts

```diff
@@ -0,0 +1,38 @@
+import {ChangeDetectionStrategy, Component} from '@angular/core';
+import {NavigationEnd, Router} from '@angular/router';
+import {filter} from 'rxjs/operators';
+import {environment} from '../environments/environment';
+
+interface Version {
+  text: string;
+  url: string
+}
+
+@Component({
+  selector: 'ngbd-demo-versions',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="nav-item" ngbDropdown>
+      <a class="nav-link" ngbDropdownToggle id="demo-site-versions" role="button">
+        ng-bootstrap v{{ current }}
+      </a>
+      <div ngbDropdownMenu aria-labelledby="demo-site-versions" class="dropdown-menu dropdown-menu-end">
+        <a ngbDropdownItem *ngFor="let version of versions$ | async" href="{{ version.url }}#{{ routerUrl }}">{{ version.text }}</a>
+      </div>
+    </div>
+  `
+})
+
+export class NgbdDemoVersionsComponent {
+  current = environment.version;
+  routerUrl = '';
+  versions$: Promise<Version[]> = (window as any).NGB_DEMO_VERSIONS;
+
+  constructor(router: Router) {
+    router.events.pipe(
+      filter(event => event instanceof NavigationEnd)
+    ).subscribe((event: NavigationEnd) => {
```

Comment:
I know the component is supposed to be always present, but in case it changes in the future, wouldn't be better to have a correct clean-up mechanism to unsubscribe if the component is destroyed?

Reply:
Yep, I haven't add it explicitly because it is always present. I really doubt it will change, as the idea for this component is to be always visible. I still can add the cleanup code, sure.

Reply:
fixed